src/services/transports/transports-logics/load-balance.ts

```diff
@@ -0,0 +1,85 @@
+import { createTransport, type Transport } from 'viem';
+import { LoadBalanceConfig } from '../types';
+
+type RPCMetrics = {
+  avgProcessingTime: number;
+  successRate: number;
+  pending: number;
+  samples: number;
+};
+
+type MethodMetrics = Record<string, RPCMetrics>;
+
+export function loadBalance(transports: readonly Transport[], config: LoadBalanceConfig = {}): Transport {
+  const { key = 'loadbalance', name = 'Load Balancing', minSuccessRate = 0.05, minSamples = 3, maxAttempts = 5 } = config;
+
+  const rpcMetrics: MethodMetrics[] = transports.map(() => ({}));
+
+  return ({ chain, timeout, ...rest }) => {
+    const transportInstances = transports.map((t) => t({ chain, timeout, ...rest }));
+
+    return createTransport({
+      key,
+      name,
+      async request({ method, params }) {
+        const processRequest = async (index: number): Promise<any> => {
+          // If we have tried all transports or reached the max attempts, throw an error
+          if (index >= transportInstances.length || (maxAttempts && index >= maxAttempts)) {
+            throw new Error('All RPC attempts failed');
+          }
+
+          const transport = transportInstances[index];
+
+          // Get the metrics for the method
+          const metrics = rpcMetrics[index][method] || { avgProcessingTime: 0, successRate: 1, pending: 0, samples: 0 };
+
+          // A min amount of samples to discard an RPC because of the success rate
+          if (metrics.samples >= minSamples && metrics.successRate < minSuccessRate) {
+            return processRequest(index + 1);
```

Comment:
If you do this then you won't be sorting by best again, right?
May I suggest something similar to what the [indexer does](https://github.com/Balmy-protocol/indexer/blob/dev/src/components/resources/generic/load-balancer-resource.ts#L61)? Every time you get a request you copy the array of transports, and then you sort them by score. Then you try the best one. If it fails, you remove it from the array and re-sort and so on